src/txn/txn_rollback_to_stable.c

```diff
@@ -454,8 +466,32 @@ __txn_rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
         if (!WT_PREFIX_MATCH(uri, "file:"))
             continue;
 
+        WT_ERR(cursor->get_value(cursor, &config));
+
+        /* Find out the max durable timestamp of the object from checkpoint. */
+        newest_durable_ts = WT_TS_NONE;
+        durable_ts_found = false;
+        WT_ERR(__wt_config_getones(session, config, "checkpoint", &cval));
+        __wt_config_subinit(session, &ckptconf, &cval);
+        for (; __wt_config_next(&ckptconf, &key, &cval) == 0;) {
+            ret = __wt_config_subgets(session, &cval, "newest_durable_ts", &durableval);
+            WT_ERR_NOTFOUND_OK(ret);
+            if (ret == 0) {
+                newest_durable_ts = WT_MAX(newest_durable_ts, (wt_timestamp_t)durableval.val);
+                durable_ts_found = true;
+            }
+        }
+
         WT_ERR(__wt_session_get_dhandle(session, uri, NULL, NULL, 0));
-        WT_SAVE_DHANDLE(session, ret = __txn_rollback_to_stable_btree(session));
+        /*
+         * The rollback operation should be performed on this file based on the following:
+         * 1. There is no durable timestamp in any checkpoint.
+         * 2. The checkpoint durable timestamp is greater than the rollback timestamp.
+         * 3. The tree is modified.
+         */
+        if (!durable_ts_found || newest_durable_ts > rollback_timestamp || S2BT(session)->modified)
+            WT_SAVE_DHANDLE(
+              session, ret = __txn_rollback_to_stable_btree(session, rollback_timestamp));
```

Comment:
I'm confused about calling `WT_SAVE_DHANDLE` here. If we set `WT_SESSION.dhandle` at line 485, why do we need to save the dhandle when calling `__txn_rollback_to_stable_btree()`?

I'm guessing the caller of `__txn_rollback_to_stable_btree_apply()` might reasonably save the dhandle, but maybe I'm missing something.

Reply:
Yes, there is no need of `WT_SAVE_HANDLE`. I removed it.